modules/logging/src/main/scala/com/evolutiongaming/skafka/producer/LoggingProducer.scala

```diff
@@ -1,51 +1,48 @@
 package com.evolutiongaming.skafka.producer
 
-import com.evolutiongaming.concurrent.CurrentThreadExecutionContext
-import com.evolutiongaming.safeakka.actor.ActorLog
+import cats.implicits._
+import cats.{Monad, MonadError}
 import com.evolutiongaming.skafka.{OffsetAndMetadata, ToBytes, Topic, TopicPartition}
 
-import scala.concurrent.Future
 import scala.concurrent.duration.FiniteDuration
-import scala.util.{Failure, Success}
 
 object LoggingProducer {
+  type CanFail[F[_]] = MonadError[F, Throwable]
 
-  def apply(producer: Producer[Future], log: ActorLog): Producer[Future] = {
-    implicit val ec = CurrentThreadExecutionContext
+  def apply[F[_] : CanFail : Monad](producer: Producer[F], log: Log[F]): Producer[F] = {
 
-    new Producer[Future] {
+    new Producer[F] {
 
-      def initTransactions() = producer.initTransactions()
+      override val initTransactions = producer.initTransactions
```

Comment:
do we really need `override` ?

Reply:
It has been my belief that adding `override` when applicable is some good habit. E.g. it makes line-specific errors be generated when implementation's signature is unintentionally changed.

Reply:
ok I'm fine with that, however I'm not among those "believers" :)